doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -133,81 +116,84 @@ Programmable proof search
          Goal forall P:nat -> Prop, P 0 -> exists n, P n.
          eauto.
 
-      Note that ``ex_intro`` should be declared as a hint.
+      `ex_intro` is declared as a hint so the proof succeeds.
 
+   .. seealso:: :ref:`thehintsdatabasesforautoandeauto`
 
-   .. tacv:: {? info_}eauto {? @natural} {? using {+ @qualid}} {? with {+ @ident}}
+   .. tacn:: info_eauto {? @int_or_var } {? @auto_using } {? @hintbases }
 
-      The various options for :tacn:`eauto` are the same as for :tacn:`auto`.
+      The various options for :tacn:`info_eauto` are the same as for :tacn:`info_auto`.
 
    :tacn:`eauto` also obeys the following flags:
 
    .. flag:: Info Eauto
              Debug Eauto
       :undocumented:
 
-   .. seealso:: :ref:`The Hints Databases for auto and eauto <thehintsdatabasesforautoandeauto>`
-
+   .. tacn:: debug eauto {? @int_or_var } {? @auto_using } {? @hintbases }
 
-.. tacn:: autounfold with {+ @ident}
-   :name: autounfold
+      Behaves like :tacn:`eauto` but shows the tactics it tries to solve the goal,
+      including failing paths.
 
-   This tactic unfolds constants that were declared through a :cmd:`Hint Unfold`
-   in the given databases.
+   .. tacn::bfs eauto
 
-.. tacv:: autounfold with {+ @ident} in @goal_occurrences
+      Like :tacn:`eauto`, but does a breadth-first search (i.e. the tactic
+      checks all the relevant hints to see if one solves the goal before
+      recurring).
 
-   Performs the unfolding in the given clause (:token:`goal_occurrences`).
+.. tacn:: autounfold {? @hintbases } {? @occurrences }
 
-.. tacv:: autounfold with *
+   Unfolds constants that were declared through a :cmd:`Hint Unfold`
+   in the given databases.
 
-   Uses the unfold hints declared in all the hint databases.
+   :n:`@occurrences`
+     Performs the unfolding in the specified occurrences.  The :n:`at @occs_nums`
+     clause of :n:`@occurrences` is supported.
 
-.. tacn:: autorewrite with {+ @ident}
-   :name: autorewrite
+.. tacn:: autorewrite {? * } with {+ @ident } {? @occurrences } {? using @ltac_expr }
 
-   This tactic carries out rewritings according to the rewriting rule
-   bases :n:`{+ @ident}`.
+   `*`
+     If present, rewrite all occurrences.
 
-   Each rewriting rule from the base :n:`@ident` is applied to the main subgoal until
-   it fails. Once all the rules have been processed, if the main subgoal has
-   progressed (e.g., if it is distinct from the initial main goal) then the rules
-   of this base are processed again. If the main subgoal has not progressed then
-   the next base is processed. For the bases, the behavior is exactly similar to
-   the processing of the rewriting rules.
+   :n:`with {+ @ident }`
+     Specifies the rewriting rule bases to use.
 
-   The rewriting rule bases are built with the :cmd:`Hint Rewrite`
-   command.
+   :n:`@occurrences`
+     Performs rewriting in the specified occurrences.  Note: the `at` clause on
+     the conclusion is currently not supported.
```

Comment:
I would remove "on the conclusion" from here because even if using an `at` clause for an hypothesis won't trigger an error, it will not change the behavior either:

```coq
Parameter a b : nat.
Axiom ax : a = b.
Hint Rewrite ax : foo.
Goal a = a -> a = a.
intro.
autorewrite with foo in H at 2 |-.
(* rewrites both occurrences in H *)
```

Reply:
We should make `occurrences` work the same in all tactics (to the extent possible).  I would expect that's not technically difficult but requires changing a number of tactic and sharing more code between them.